packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -347,22 +343,38 @@ export default class ScriptTransformer {
       }
     }
 
+    // Apply instrumentation to the code if necessary, keeping the instrumented code and new map
+    let map = transformed.map;
     if (!transformWillInstrument && instrument) {
-      code = this._instrumentFile(
+      /**
+       * We can map the original source code to the instrumented code ONLY if
+       * - the process of transforming the code produced a source map e.g. ts-jest
+       * - we did not transform the source code
+       *
+       * Otherwise we cannot make any statements about how the instrumented code corresponds to the original code,
+       * and we should NOT emit any source maps
+       *
+       */
+      const shouldEmitSourceMaps = (!!transform && !!map) || !transform;
```

Comment:
```suggestion
      const shouldEmitSourceMaps = (transform != null && map != null) || transform == null;
```

all the `!` in a row confused me 😅

Reply:
Applied this suggestion.